src/bun.js/bindings/sqlite/JSSQLStatement.cpp

```diff
@@ -671,63 +806,174 @@ static inline bool rebindValue(JSC::JSGlobalObject* lexicalGlobalObject, sqlite3
+                    return target->getDirectIndex(globalObject, integer.value() - 1);
+                }
+            }
+
+            const auto identifier = Identifier::fromString(vm, str);
+            PropertySlot slot(target, PropertySlot::InternalMethodType::GetOwnProperty);
+            if (!target->getOwnNonIndexPropertySlot(vm, &structure, identifier, slot)) {
+                return JSValue();
+            }
+
+            if (LIKELY(!slot.isTaintedByOpaqueObject())) {
+                return slot.getValue(globalObject, identifier);
+            }
+
+            return target->get(globalObject, identifier);
+        };
+
+        for (size_t i = 0; i < size; i++) {
+            auto* name = sqlite3_bind_parameter_name(stmt, i + 1);
+
+            JSValue value = getValue(name, i);
+            if (!value && !scope.exception()) {
+                if (throwOnMissing) {
+                    throwException(globalObject, scope, createError(globalObject, makeString("Missing parameter \""_s, name, "\""_s)));
+                } else {
+                    continue;
+                }
+            }
             RETURN_IF_EXCEPTION(scope, JSValue());
+
+            if (!rebindValue(globalObject, db, stmt, i + 1, value, scope, clone, safeIntegers)) {
+                return JSValue();
+            }
+
+            RETURN_IF_EXCEPTION(scope, {});
+            count++;
+        }
+    }
+    // Does it only contain indexed properties?
+    //
+    // { 0: "foo", 1: "bar", "2": "baz" }
+    //
+    else if (UNLIKELY(bindings.isOnlyIndexed)) {
+        for (size_t i = 0; i < size; i++) {
+            JSValue value = target->getDirectIndex(globalObject, i);
+            if (!value && !scope.exception()) {
+                if (throwOnMissing) {
+                    throwException(globalObject, scope, createError(globalObject, makeString("Missing parameter \""_s, i + 1, "\""_s)));
+                } else {
+                    continue;
+                }
+            }
+
+            RETURN_IF_EXCEPTION(scope, JSValue());
+
+            if (!rebindValue(globalObject, db, stmt, i + 1, value, scope, clone, safeIntegers)) {
+                return JSValue();
+            }
+
+            RETURN_IF_EXCEPTION(scope, {});
+            count++;
         }
+    }
+    // Is it a simple object with no getters or setters?
+    //
+    // { foo: "bar", baz: "qux" }
+    //
+    else if (target->canUseFastGetOwnProperty(structure)) {
+        for (size_t i = 0; i < size; i++) {
+            const auto& property = bindingNames[i];
+            JSValue value = property.isEmpty() ? target->getDirectIndex(globalObject, i) : target->fastGetOwnProperty(vm, structure, bindingNames[i]);
+            if (!value && !scope.exception()) {
+                if (throwOnMissing) {
+                    throwException(globalObject, scope, createError(globalObject, makeString("Missing parameter \""_s, property.isEmpty() ? String::number(i) : property.string(), "\""_s)));
```

Comment:
same here

Reply:
It checks two lines below

https://github.com/oven-sh/bun/blob/0ba290c5395b6d9393a1069b68d4d1aae64ae259/src/bun.js/bindings/sqlite/JSSQLStatement.cpp#L922